edi/models/edi_backend.py

```diff
@@ -198,34 +206,88 @@ def _exchange_output_check(self, exchange_record):
-        type_code = exchange_record.exchange_type_id.code
-        return [
-            # specific for backend type and exchange type
-            base_usage + "." + type_code,
-            # specific for backend type
-            base_usage,
-        ]
+            return component.send()
+        raise NotImplementedError("No handler for `_exchange_send`")
 
-    def _exchange_notify_record(self, record, message, level="info"):
+    def _exchange_notify_record(self, exchange_record, message, level="info"):
         """Attach notification of exchange state to the original record."""
-        if not hasattr(record.record, "message_post_with_view"):
+        if not hasattr(exchange_record.record, "message_post_with_view"):
             return
-        record.record.message_post_with_view(
+        exchange_record.record.message_post_with_view(
             "edi.message_edi_exchange_link",
             values={
                 "backend": self,
-                "exchange_record": record,
+                "exchange_record": exchange_record,
                 "message": message,
                 "level": level,
             },
             subtype_id=self.env.ref("mail.mt_note").id,
         )
+
+    def _cron_check_output_exchange_sync(self, **kw):
+        for backend in self:
+            backend._check_output_exchange_sync(**kw)
+
+    def _check_output_exchange_sync(self, skip_send=False):
+        """Lookup for pending output records and take care of them.
+
+        First work on records that need output generation.
+        Then work on records waiting for a state update.
+
+        :param skip_send: only generate missing output.
+        """
+        # Generate output files
+        new_records = self.exchange_record_model.search(
+            self._output_new_records_domain()
+        )
+        _logger.info(
+            "EDI Exchange output sync: " "found %d new records to process.",
+            len(new_records),
+        )
+        for rec in new_records:
+            self.generate_output(rec)
+
+        if skip_send:
+            return
+        pending_records = self.exchange_record_model.search(
+            self._output_pending_records_domain()
+        )
+        _logger.info(
+            "EDI Exchange output sync: " "found %d pending records to process.",
+            len(pending_records),
+        )
+        for rec in pending_records:
+            if rec.edi_exchange_state == "output_pending":
+                self.exchange_send(rec)
+            else:
+                self._exchange_output_check_state(rec)
+
+    def _output_new_records_domain(self):
+        return [
+            ("type_id.exchange_file_auto_generate", "=", True),
+            ("type_id.direction", "=", "output"),
+            ("edi_exchange_state", "=", "new"),
+            ("exchange_file", "=", False),
+        ]
+
+    def _output_pending_records_domain(self):
+        states = ("output_pending", "output_sent", "output_sent_and_error")
```

Comment:
Why output sent should be pending?

I would also add a check on the type in order to proceed with this. What do you think?

Reply:
because `_check_output_exchange_sync` takes care of sending them as well

Reply:
> I would also add a check on the type in order to proceed with this.

what do you mean? The type's direction is "output", what else?

Reply:
Maybe the output will be sent on user decision, not by a cron

It is just an idea (similar to output generation??)

Reply:
hmm yeah, why not... 
I could also split sending from sent here

Reply:
We can keep it for version two, so, don't worry :smile: